src/CoreFoundation/CFProxySupport.cs

```diff
@@ -580,61 +580,105 @@ public static CFProxySettings GetSystemProxySettings ()
 			return new CFProxySettings (dict);
 		}
 		
-#if notyet
-		// FIXME: These require bindings for CFRunLoopSource and CFStreamClientContext
-		
-		public delegate void CFProxyAutoConfigurationResultCallback (IntPtr client, NSArray proxyList, NSError error);
+		delegate void CFProxyAutoConfigurationResultCallback (IntPtr client, IntPtr proxyList, IntPtr error);
 		
 		[DllImport (Constants.CFNetworkLibrary)]
 		extern static /* CFRunLoopSourceRef __nonnull */ IntPtr CFNetworkExecuteProxyAutoConfigurationScript (
 			/* CFStringRef __nonnull */ IntPtr proxyAutoConfigurationScript,
 			/* CFURLRef __nonnull */ IntPtr targetURL,
-			/* CFProxyAutoConfigurationResultCallback __nonnull */ IntPtr cb,
-			/* CFStreamClientContext * __nonnull */ IntPtr clientContext);
+			/* CFProxyAutoConfigurationResultCallback __nonnull */ CFProxyAutoConfigurationResultCallback cb,
+			/* CFStreamClientContext * __nonnull */ ref CFStreamClientContext  clientContext);
 		
-		public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (NSString proxyAutoConfigurationScript, NSUrl targetURL, CFProxyAutoConfigurationResultCallback resultCallback, CFStreamClientContext clientContext)
+		public static CFProxy[] ExecuteProxyAutoConfigurationScript (string proxyAutoConfigurationScript, Uri targetUrl)
```

Comment:
While this overload is fine by itself, I think we should also provide a more advanced version that returns the CFRunLoopSource, so that users have more control over exactly which run loop to use, in particular because it's too easy to end up doing network requests on the main thread with this implementation, since it will execute synchronously on the current thread (which will most likely be the main thread).

Something like this maybe:

```csharp
public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (string script, Uri targetUrl, Action<CFProxy[], NSError> callback)
{
    // ...
}
```

another possible variation would be an async version:

```csharp
public static Task<CFProxy[]> ExecuteProxyAutoConfigurationScriptAsync (string script, Uri targetUrl)
{
    // ...
}
```

which schedules the runloop on a bacground/threadpool thread.

And then the same for `ExecuteProxyAutoConfigurationUrl` of course.

Reply:
I'll provide both options, Task and CFRunLoopSource